Sources/NIOHTTPCompression/HTTPResponseCompressor.swift

```diff
@@ -66,6 +73,42 @@ public final class HTTPResponseCompressor: ChannelDuplexHandler, RemovableChanne
     public typealias OutboundIn = HTTPServerResponsePart
     /// This class emits `HTTPServerResponsePart` outbound.
     public typealias OutboundOut = HTTPServerResponsePart
+    
+    /// A closure that accepts a response header, optionally modifies it, and returns `true` if the response it belongs to should be compressed.
+    ///
+    /// - Parameter responseHeaders: The headers that will be used for the response. These can be modified as needed at this stage, to clean up any marker headers used to statelessly determine if compression should occur, and the new headers will be used when writing the response. Compression headers are not yet provided and should not be set; ``HTTPResponseCompressor`` will set them accordingly based on the result of this predicate.
+    /// - Parameter isCompressionSupported: Set to `true` if the client requested compatible compression, and if the HTTP response supports it, otherwise `false`.
+    /// - Returns: Return ``CompressionIntent/compressibleResponse`` if the compressor should proceed to compress the response, or ``CompressionIntent/uncompressedResponse`` if the response should not be compressed.
+    ///
+    /// - Note: Returning ``CompressionIntent/compressibleResponse`` is only a suggestion — when compression is not supported, the response will be returned as is along with any modified headers.
+    public typealias ResponseCompressionPredicate = (
+        _ responseHeaders: inout HTTPResponseHead,
+        _ isCompressionSupported: Bool
+    ) -> CompressionIntent
+    
+    /// A signal a ``ResponseCompressionPredicate`` returns to indicate if it intends for compression to be used or not when supported by HTTP.
+    public struct CompressionIntent: Sendable, Hashable {
+        /// The internal type ``CompressionIntent`` uses.
+        enum RawValue {
+            /// The response should be compressed if supported by the HTTP protocol.
+            case compressibleResponse
+            /// The response should not be compressed even if supported by the HTTP protocol.
+            case uncompressedResponse
+        }
+        
+        /// The raw value of the intent.
+        let rawValue: RawValue
+        
+        /// Initialize the raw value with an internal intent.
+        init(_ rawValue: RawValue) {
+            self.rawValue = rawValue
+        }
+        
+        /// The response should be compressed if supported by the HTTP protocol.
+        public static let compressibleResponse = CompressionIntent(.compressibleResponse)
+        /// The response should not be compressed even if supported by the HTTP protocol.
+        public static let uncompressedResponse = CompressionIntent(.uncompressedResponse)
```

Comment:
I think let's make it clear these are action requests:

```
.compressIfPossible
.doNotCompress
```

Reply:
Updated!